js/react_native/android/src/main/cpp/cpp-adapter.cpp

```diff
@@ -0,0 +1,130 @@
+#include <jni.h>
+#include <jsi/jsi.h>
+#include <string>
+// #include <android/log.h>
+
+using namespace facebook;
+
+typedef u_int8_t byte;
+
+std::string jstring2string(JNIEnv *env, jstring jStr) {
+  if (!jStr) return "";
+
+  jclass stringClass = env->GetObjectClass(jStr);
+  jmethodID getBytes = env->GetMethodID(stringClass, "getBytes", "(Ljava/lang/String;)[B");
+  const auto stringJbytes = (jbyteArray) env->CallObjectMethod(jStr, getBytes, env->NewStringUTF("UTF-8"));
+
+  auto length = (size_t) env->GetArrayLength(stringJbytes);
+  jbyte* pBytes = env->GetByteArrayElements(stringJbytes, nullptr);
+
+  std::string ret = std::string((char *)pBytes, length);
+  env->ReleaseByteArrayElements(stringJbytes, pBytes, JNI_ABORT);
+
+  env->DeleteLocalRef(stringJbytes);
+  env->DeleteLocalRef(stringClass);
+  return ret;
+}
+
+extern "C"
+JNIEXPORT void JNICALL
+Java_ai_onnxruntime_reactnative_OnnxruntimeJSIHelper_nativeInstall(JNIEnv *env, jclass _, jlong jsiPtr, jobject instance) {
+  auto jsiRuntime = reinterpret_cast<jsi::Runtime*>(jsiPtr);
+
+  auto& runtime = *jsiRuntime;
+
+  auto instanceGlobal = env->NewGlobalRef(instance);
+  auto getBytesFromBlob = [=](const std::string& blobId, int offset, int size) -> byte* {
+    if (!env) throw std::runtime_error("JNI Environment is gone!");
+
+    // get java class
+    jclass clazz = env->GetObjectClass(instanceGlobal);
+    // get method in java class
+    jmethodID getBufferJava = env->GetMethodID(clazz, "getBlobBuffer", "(Ljava/lang/String;II)[B");
+    // call method
+    auto jstring = env->NewStringUTF(blobId.c_str());
+    auto boxedBytes = (jbyteArray) env->CallObjectMethod(instanceGlobal,
+                                                          getBufferJava,
+                                                          // arguments
+                                                          jstring,
+                                                          offset,
+                                                          size);
+    env->DeleteLocalRef(jstring);
+
+    jboolean isCopy = true;
+    jbyte* bytes = env->GetByteArrayElements(boxedBytes, &isCopy);
+    env->DeleteLocalRef(boxedBytes);
+    return reinterpret_cast<byte*>(bytes);
+  };
+
+  auto resolveArrayBuffer = jsi::Function::createFromHostFunction(runtime,
+                                                                  jsi::PropNameID::forAscii(runtime, "getArrayBufferForBlob"),
```

Comment:
> "getArrayBufferForBlob",

nit: looks like the function is called `resolveArrayBuffer`,  should we update this name here?